src/main/java/proai/cache/RecordCache.java

```diff
@@ -353,15 +353,16 @@ private SchemaLocator createLocator(File schemaDir) throws Exception {
     }
 
     private static void addToCatalog(SchemaCatalog catalog, String url, String path) throws Exception {
-
         if (!catalog.contains(url)) {
-            InputStream in = null;
-            try {
-                in = RecordCache.class.getClassLoader().getResourceAsStream(path);
-            } catch (Throwable th) {
-                in = ClassLoader.getSystemResourceAsStream(path);
+            InputStream in;
+            if (
+                    (in = RecordCache.class.getClassLoader().getResourceAsStream(path)) != null ||
+                    (in = ClassLoader.getSystemResourceAsStream(path)) != null) {
```

Comment:
I don't remember. This was likely some code I copied from Kowari (now Mulgara: https://github.com/quoll/mulgara) at the time. I recall it doing something similar when loading resources, and figured it was more robust, but doubt it's actually useful here.